src/test/ui/span/macro-span-replacement.stderr

```diff
@@ -1,11 +1,8 @@
 warning: struct is never constructed: `S`
-  --> $DIR/macro-span-replacement.rs:7:14
+  --> $DIR/macro-span-replacement.rs:7:12
    |
 LL |         $b $a;
-   |              ^
-...
-LL |     m!(S struct);
-   |     ------------- in this macro invocation
+   |            ^^
```

Comment:
This is a error quality degradation.

Reply:
Mmm. I was looking at this before I submit the PR, and was a bit confused as to why my changes would impact this.

Now that I've had some time to get more familiar with the rust code base, I see that here:

https://github.com/rust-lang/rust/blob/master/src/librustc_errors/emitter.rs#L354-L366

We check if the span is inside of a macro.  Previously, on this code, `def_span(item.span)` would just return `item.span` (because it didn't find a {), which *was* considered inside of a macro (not sure why?), whereas ident.span is not conisdered inside a macro. 

Not sure how to resolve this, tbh.  Open to advice from people with more experience with the code base!

Reply:
I believe that the `item.span` was marked with `with_ctxt`, whereas this new span wasn't, because in the current model it is true that this span is fully enclosed by the macro and has no other context. I would merge this PR as is with a follow up ticket filed to address this case. It might be a good idea to mark all the spans that involve a macro expansion site with a context so that these type of errors highlight the caller as well. That change would likely cause multiple other changes in other errors.

Reply:
@estebank roger, opened a different issue for it.